src/dbnode/storage/bootstrap/run_state.go

```diff
@@ -0,0 +1,102 @@
+package bootstrap
+
+import (
+	"errors"
+
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/persist"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+)
+
+var (
+	errFilesystemOptsNotSet   = errors.New("filesystemOptions not set")
+	errInfoFilesFindersNotSet = errors.New("infoFilesFinders not set")
+)
+
+// NewRunState creates state specifically to be used during the bootstrap process.
+// Primarily a mechanism for passing info files along without needing to re-read them at each
+// stage of the bootstrap process.
+func NewRunState(options RunStateOptions) (RunState, error) {
+	// TODO(nate): enable this once properly instantiating in NamespacesTester in util.go
+	// if err := options.Validate(); err != nil {
+	//   return RunState{}, err
+	// }
+	return RunState{
+		fsOpts:           options.FilesystemOptions(),
+		infoFilesFinders: options.InfoFilesFinders(),
+	}, nil
+}
+
+// InfoFilesForNamespace returns the info files grouped by shard for the provided namespace.
+func (r *RunState) InfoFilesForNamespace(ns namespace.Metadata) InfoFileResultsPerShard {
+	return r.ReadInfoFiles()[ns]
+}
+
+// TODO(nate): Make this threadsafe? If so, we'll need to clone the map
```

Comment:
We always call this in a singlethreaded fashion, but maybe add some synchronization controls here just in case? Thoughts @robskillington @notbdu ?